api/src/main/java/org/ehcache/spi/persistence/StateRepository.java

```diff
@@ -18,11 +18,30 @@
 
 import java.io.Serializable;
 
+import static java.util.Arrays.asList;
+
 /**
  * A repository allowing to preserve state in the context of a {@link org.ehcache.Cache}.
  */
 public interface StateRepository {
 
+  /**
+   * Gets a named state holder rooted in the current {@code StateRepository}.
+   * <p>
+   * If the state holder existed already, it is returned with its content fully available.
+   *
+   * @param name       the state holder name
+   * @param keyClass   concrete key type
+   * @param valueClass concrete value type
+   * @param <K>        the key type, must be {@code Serializable}
+   * @param <V>        the value type, must be {@code Serializable}
+   * @param whiteListedClasses classes whose objects would get deserialized as part of key or value deserialization
+   * @return a state holder
+   */
+  default <K extends Serializable, V extends Serializable> StateHolder<K, V> getPersistentStateHolder(String name, Class<K> keyClass, Class<V> valueClass, Class ... whiteListedClasses) {
```

Comment:
You should be using the wildcard `Class<?>` here instead of the raw type - same goes for every other Class reference added.

Reply:
done